docs/developing.rst

```diff
@@ -571,6 +571,25 @@ Example::
 
             toil.exportFile(outputFileID, 'file:///some/other/local/path')
 
+
+There are some trade-offs to file staging that should be considered. It may be more efficient to
+directly download a file for cases in which the cost of the copy operation imposed by an import is
+unnecessary. For example, a download is better for large files that are used only once. While
+importing is more efficient if the file in question is needed several times, provided that the
+storage platform of the job store supports certain optimizations gained by the import. For example,
+Amazon's S3 platform, used in the AWS job store, is better optimized for reads and writes to files
+that are in the same region. Thus an import is justified if a source file is in a different region
+than that of the job store. Note that toil-scripts/lib contains common functionality for downloading
+URLs (see `https://github.com/BD2KGenomics/toil-scripts/blob/master/src/toil_scripts/lib/urls.py
```

Comment:
Use the separate syntax, where the target URL is listed below the paragraph. See http://www.sphinx-doc.org/en/stable/rest.html#external-links.

Also, the text `toil-scripts/lib` should be what's clickable, not the URL itself. The reader should only see the URL when hovering over the link, or after clicking on it.